monai/utils/module.py

```diff
@@ -213,11 +213,15 @@ def instantiate(path: str, **kwargs):
     component = locate(path)
     if component is None:
         raise ModuleNotFoundError(f"Cannot locate class or function path: '{path}'.")
-    if isclass(component):
-        return component(**kwargs)
-    # support regular function, static method and class method
-    if isfunction(component) or (ismethod(component) and isclass(getattr(component, "__self__", None))):
-        return partial(component, **kwargs)
+    try:
+        if isclass(component):
+            return component(**kwargs)
+        # support regular function, static method and class method
+        if isfunction(component) or (ismethod(component) and isclass(getattr(component, "__self__", None))):
+            return partial(component, **kwargs)
+    except BaseException as e:
+        warnings.warn(f"\n----\nCannot instantiate '{path}' with kwargs: {kwargs}\n{e}\n----\n")
```

Comment:
Should we add this warning message to the exception error `e` directly?
I think it's already part of the error message.

Thanks.